### PR TITLE
React to lobby kick / close game request from server

### PR DESCRIPTION
**Pull request by Geosearchef**

### CODE REVIEW

File: src/main/java/com/faforever/client/FafClientApplication.java [hunk 1]
```diff
@@ -120,5 +122,30 @@ private void showMainWindow() {
   public void stop() throws Exception {
     applicationContext.close();
     super.stop();
+
+    Thread stopThread = new Thread(() -> {
+      try {
+        Thread.sleep(Duration.ofSeconds(20).toMillis());
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/FafClientApplication.java]: Rather 30 seconds

File: src/main/java/com/faforever/client/FafClientApplication.java [hunk 2]
```diff
@@ -120,5 +122,30 @@ private void showMainWindow() {
...
+      } catch (InterruptedException e) {
+      }
+
+      Thread.getAllStackTraces()
+          .entrySet()
+          .stream()
+          .filter(e -> !e.getKey().isDaemon())
+          .forEach(e -> {
+            log.info("Non daemon Thread \"{}\" (id: {}) still active in state: {}", e.getKey().getName(), e.getKey().getId(), e.getKey().getState());
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/FafClientApplication.java]: warn

**Review by 1-alex98 — DISMISSED** [review]

**Comment by 1-alex98** [comment]
pls rebase

File: src/main/java/com/faforever/client/FafClientApplication.java [hunk 3]
```diff
@@ -120,5 +122,30 @@ private void showMainWindow() {
...
+      }
+
+      Thread.getAllStackTraces()
+          .entrySet()
+          .stream()
+          .filter(e -> !e.getKey().isDaemon())
+          .forEach(e -> {
+            log.error("Non daemon Thread \"{}\" (id: {}) still active in state: {}", e.getKey().getName(), e.getKey().getId(), e.getKey().getState());
+            log.error("Stacktrace of thread {}:\n{}", e.getKey().getName(), Arrays.stream(e.getValue()).map(Object::toString).collect(Collectors.joining("\n")));
+          });
```
Comment (Katharsas) [inline src/main/java/com/faforever/client/FafClientApplication.java]: Maybe it should collect the non-daemon threads and if the size is 0 it should return normally. I know that daemon threads are supposed to terminate while in-process but maybe they get interrupted first and then you could just return.

**Review by Katharsas — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/FafClientApplication.java (hunk 3 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/FafClientApplication.java]: Yeah maybe could be. Tho chances are small that happens i think. Even tho not sure if it happens at all

Comment (Geosearchef) [reply src/main/java/com/faforever/client/FafClientApplication.java]: done

**Comment by Geosearchef** [comment]
rebased

**Review by 1-alex98 — DISMISSED** [review]

**Comment by Katharsas** [comment]
Im confused, why does this PR exist twice? (https://github.com/FAForever/downlords-faf-client/pull/1481)

File: src/main/java/com/faforever/client/game/GameService.java [hunk 4]
```diff
@@ -655,4 +656,16 @@ private void removeGame(GameInfoMessage gameInfoMessage) {
     }
     eventBus.post(new GameRemovedEvent(game));
   }
+
+  public void stopGame() {
```
Comment (Katharsas) [inline src/main/java/com/faforever/client/game/GameService.java]: I would give it a stronger name like `killGame` or something.

**Comment by Askaholic** [comment]
Can we please localize the kick message on the client instead of having it hardcoded somewhere deep inside the lobby server.

https://github.com/FAForever/server/issues/499

**Comment by Geosearchef** [comment]
@Askaholic :+1: will you change it server sided?

**Comment by Askaholic** [comment]
For sure! I'll also do it for any other message that's currently being sent in plain English by the server. 

Client issue: #1292
Server issue with list of messages that need work: FAForever/server#444.

**Comment by Askaholic** [comment]
@Geosearchef Done https://github.com/FAForever/server/pull/502

**Comment by Geosearchef** [comment]
done

File: src/main/java/com/faforever/client/FafClientApplication.java [hunk 5]
```diff
@@ -120,5 +124,34 @@ private void showMainWindow() {
...
+
+      threads.stream()
+          .filter(e -> !e.getKey().isDaemon())
+          .forEach(e -> {
+            log.error("Non daemon Thread \"{}\" (id: {}) still active in state: {}", e.getKey().getName(), e.getKey().getId(), e.getKey().getState());
+            log.error("Stacktrace of thread {}:\n{}", e.getKey().getName(), Arrays.stream(e.getValue()).map(Object::toString).collect(Collectors.joining("\n")));
+          });
+
+      try {
+        Thread.sleep(Duration.ofSeconds(1).toMillis());
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/FafClientApplication.java:148]: why wait?

**Review by 1-alex98 — DISMISSED** [review]

**Comment by 1-alex98** [comment]
@Geosearchef pls in the future issue and "fixes #" both are missing here and now it is not in the change log , same goes for some other Prs of yours

**Comment by 1-alex98** [comment]
Always have an issue
